src/sessions.ts

```diff
@@ -123,15 +129,23 @@ class ClientSession extends TypedEventEmitter<ClientSessionEvents> {
 
     options = options ?? {};
 
+    if (options.snapshot === true) {
+      this.snapshot = true;
+      if (options.causalConsistency === true) {
+        // TODO: throw validation error; MongoDriverError or more specific
+      }
+
+      // TODO(NODE-3394): also validate server version >= 5.0
```

Comment:
Use topology.capabilities (with a new method) to check this

Reply:
implemented